internal/exec/stages/files/units.go

```diff
@@ -17,37 +17,92 @@ package files
 import (
 	"fmt"
 	"path/filepath"
+	"strconv"
 	"strings"
 
+	"github.com/coreos/go-systemd/dbus"
 	"github.com/coreos/ignition/v2/config/v3_1_experimental/types"
 	"github.com/coreos/ignition/v2/internal/distro"
 	"github.com/coreos/ignition/v2/internal/exec/util"
 )
 
+type Preset struct {
+	unit            string
+	enabled         bool
+	instantiateable bool
+	instances       []string
+}
+
+// warnOnOldSystemdVersion checks the version of Systemd
+// in a given system and prints a warning if older than 240.
+func (s *stage) warnOnOldSystemdVersion() error {
+	var systemdVersion string
+	var begin, end int
+	conn, err := dbus.New()
+	if err != nil {
+		return err
+	}
+	version, err := conn.GetManagerProperty("Version")
+	if err != nil {
+		return err
+	}
+	// Handle different systemd versioning schemes that are being returned.
+	// for e.g:
+	// - Fedora 31: `v243.5-1.fc31`
+	// - RHEL 8: `239`
+	if strings.HasPrefix(version, "v") {
+		begin = 1
+
+	} else {
+		begin = 0
+	}
+	end = strings.Index(version, ".")
+	if end == -1 {
+		end = len(version)
+	}
+	systemdVersion = version[begin:end]
+	if value, err := strconv.Atoi(systemdVersion); err == nil {
```

Comment:
Up to here, you can put the fetch-and-parse logic to its own function under the `internal/systemd` package (with some parsing tests, pretty please with sugar :).

Reply:
@lucab I'm not sure how to write tests for this logic considering we're not passing any input parameters to this function.
Do you have any suggestions here?